File: APIs/samples/EVALUATION_APIs/ScanerAPI/EVAL_LONGITUDINAL_CTRL/python/eval_longitudinal_ctrl_speed.py
#!*****************************************************************************
#* \project  : SCANeR_API                                                     *
#* \file     : EVAL_LONGITUDINAL_CTRL.py                 					  *
#* \Brief    : Radar sensor, targets detection with SCANeR API.	              *
#* \Copyright: OKTAL S.A. all rights reserved                                 *
# *****************************************************************************/
#!/usr/bin/python
import os
import inspect
from time import sleep
this_file = inspect.currentframe().f_code.co_filename
this_dir = os.path.dirname(this_file)
# to find scaner_api dll
if (os.name == 'nt'):
    os.chdir(os.path.abspath(os.environ['STUDIO_PATH']+'./SCANeRstudio_1.9/APIs/bin/x64/vs2013'))

from scaner import *

parser = ScanerApiOption()
(options, args) = parser.parse_args()
Process_InitParams(options.process_name, options.configuration, ctypes.c_float(options.frequency))

status = PS_DAEMON

try:
    # write access to Shared Memory for longitudinal control
    cabToModelOutput = Com_declareOutputData('Shm/ModelCabin/CabToModel');

    while status != PS_DEAD:
        # Process manager Run 
        Process_Wait()
        Process_Run()
        
        #Process manager State
        old_status = status
        status = Process_GetState()
        
        if status == PS_RUNNING:
            Com_setShortData(cabToModelOutput, "RVLVState", 1) # RVLVState = 1 for control in speed
            Com_setDoubleData(cabToModelOutput, "RVLVTarget", 27) # target speed in m/s

        if Com_updateOutputs(UT_ShmData) == 0: #flush the corrective message
            print 'Update Shm outputs failed...'
            
except KeyboardInterrupt:
    print 'Bye bye'
    Process_Close()

File: APIs/samples/ScanerAPI/EVAL_LONGITUDINAL_CTRL/cpp/eval_longitudinal_ctrl_speed.cpp
/*!****************************************************************************
 * \project  : SCANeR_API                                                     *
 * \file     : EVAL_LONGITUDINAL_CTRL_SPEED.cpp                 					  *
 * \Brief    : Control the pedals regarding acceleration.		              *
 * \Copyright: OKTAL S.A. all rights reserved                                 *
 ******************************************************************************/
//C/C++ includes
#include <iostream>

//scaner project includes
#include "stk/Unit.h"
#include "scanerAPI/scanerAPI_DLL_C.h"
#include "scanerAPI/ScanerAPImessagesShm.h"

int main(int argc, char* argv[])
{
	Process_Init(argc, argv);

	// write access to Shared Memory for longitudinal control
	DataInterface* cabToModelOutput = Com_declareOutputData(SHM_MODELCABIN_CABTOMODEL);

	APIProcessState status = PS_DAEMON;

	while(status != PS_DEAD) // main loop
	{
		Process_Wait(); // synchronization
		Process_Run(); // step
		
		status = Process_GetState();

		if (Process_GetState() == PS_RUNNING) // simulation is running
		{
			Com_setShortData(cabToModelOutput, "RVLVState", 1); // RVLVState = 1 for control in speed
			Com_setDoubleData(cabToModelOutput, "RVLVTarget", stk::convToSI(stk::TUnitKilometrePerHour, 100)); // target speed

			Com_updateOutputs(UT_ShmData);
		}
	}
	Process_Close();
}
//--------------------------------------------------------------